cob_script_server/src/simple_script_server/simple_script_server.py

```diff
@@ -782,6 +785,85 @@ def publish_twist(self, pub, twist, end_time):
+	#
+	# # throws error code 3 in case of invalid parameter_name vector
+	def move_rel(self, component_name, parameter_name, blocking=True):
+		ah = action_handle("move_rel", component_name, parameter_name, blocking, self.parse)
+		if(self.parse):
+			return ah
+		else:
+			ah.set_active(mode="topic")
+
+		rospy.loginfo("Move <<%s>> relatively by <<%s>>", component_name, parameter_name)
+
+		# step 0: check validity of parameters:
+		if not isinstance(parameter_name, list):
+			message = "Parameter " + str(parameter_name) + " not formated correctly (not a list of numeric lists), aborting move_rel"
+			rospy.logerr(message)
+			print("parameter_name must be list")
+			ah.set_failed(3, message)
+			return ah
+		for parameter in parameter_name:
+			if not isinstance(parameter, list):
+				message = "Parameter " + str(parameter_name) + " not formated correctly (not a list of numeric lists), aborting move_rel"
+				rospy.logerr(message)
+				print("parameter_name must be list of lists")
+				ah.set_failed(3, message)
+				return ah
+			for param in parameter:
+				if not isinstance(param, (int, float)):
+					message = "Parameter " + str(parameter) + " not formated correctly (not a list of numeric lists), aborting move_rel"
+					rospy.logerr(message)
+					print("parameter_name must be list of numeric lists")
+					ah.set_failed(3, message)
+					return ah
+
+		# step 1: get current position
+		timeout = 1.0
+		try:
+			joint_names = list(rospy.wait_for_message("/" + component_name + "/joint_states", JointState, timeout = timeout).name)
+			start_pos = list(rospy.wait_for_message("/" + component_name + "/joint_states", JointState, timeout = timeout).position)
+		except rospy.ROSException as e:
+			rospy.logwarn("no joint states received from %s within timeout of %ssec. using default point time of 8sec.", component_name, str(timeout))
+			start_pos = []
+
+		# step 2: get joint limits from urdf
+		robot_urdf = URDF.from_parameter_server()
+		limits = {}
+		end_poses = []
+		for joint in robot_urdf.joints:
+			limits.update({joint.name : joint.limit})
+		
+		# step 3 calculate and send goal
+		for move_rel_param in parameter_name:
+			end_pos = []
+			if len(start_pos) == len(move_rel_param) and len(joint_names) == len(move_rel_param):
+				for i in range(len(joint_names)):
+					pos_val = start_pos[i] + move_rel_param[i]
+					# check if joint limits are exceeded
+					if limits[joint_names[i]].upper > pos_val and limits[joint_names[i]].lower < pos_val:
+						end_pos.append(pos_val)
+					else:
+						rospy.logerr("Relative motion wil exceed absolute limits! %s would go to %f but limits are: %f (upper) and %f (lower)",
+						joint_names[i], pos_val, limits[joint_names[i]].upper, limits[joint_names[i]].lower)
+						return ah
+			else:
+				rospy.logerr("Parameters don't fit with joint states!")
+				return ah
+			end_poses.append(end_pos)
+		self.move_traj(component_name, end_poses, blocking)
```

Comment:
check if `move_traj` is sucessfull and use same result for return value of `move_traj`